File: spring_webapp1/src/main/java/com/example/spring_webapp1/application/printing/ReceiptTextParser.java
package com.example.spring_webapp1.application.printing;

import com.example.spring_webapp1.domain.printing.ReceiptDocument;
import com.example.spring_webapp1.domain.printing.ReceiptLine;
import org.springframework.stereotype.Component;

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

/**
 * レシートテキスト（マーカー付き文字列）を {@link ReceiptDocument} に変換するパーサー。
 * <p>対応マーカー: [B] 太字, [L] 大字, [C] 中央, [R] 右寄せ, [LR]左テキスト|右テキスト</p>
 */
@Component
public class ReceiptTextParser {

    public ReceiptDocument parse(String text) {
        return parse(text, null);
    }

    public ReceiptDocument parse(String text, String logoPath) {
        List<ReceiptLine> lines = Arrays.stream(
                text.replace("\r\n", "\n").replace("\r", "\n").split("\n", -1)
        ).map(this::parseLine).collect(Collectors.toList());
        return new ReceiptDocument(lines, logoPath);
    }

    private ReceiptLine parseLine(String raw) {
        String s = raw;
        boolean bold = false, large = false, center = false, right = false, isLR = false;
        String leftText = "", rightText = "";

        if (s.startsWith("[B]")) { bold = true; s = s.substring(3); }
        if (s.startsWith("[L]")) { large = true; s = s.substring(3); }

        if (s.startsWith("[LR]")) {
            isLR = true;
            String rest = s.substring(4);
            int sep = rest.indexOf('|');
            leftText  = sep >= 0 ? rest.substring(0, sep).trim() : rest;
            rightText = sep >= 0 ? rest.substring(sep + 1).trim() : "";
            s = "";
        } else if (s.startsWith("[C]")) {
            center = true; s = s.substring(3);
            if (s.startsWith("[L]")) { large = true; s = s.substring(3); }
        } else if (s.startsWith("[R]")) {
            right = true; s = s.substring(3);
            if (s.startsWith("[L]")) { large = true; s = s.substring(3); }
        }

        return new ReceiptLine(s, bold, large, center, right, isLR, leftText, rightText);
    }
}

File: MauiApp1/Application/Printing/ReceiptTextParser.cs
using MauiApp1.Domain.Printing;

namespace MauiApp1.Application.Printing;

/// <summary>
/// レシートテキスト（マーカー付き文字列）を <see cref="ReceiptDocument"/> に変換するパーサー。
/// <para>対応マーカー: [B] 太字, [L] 大字, [C] 中央, [R] 右寄せ, [LR]左テキスト|右テキスト</para>
/// </summary>
public static class ReceiptTextParser
{
    private const string BoldPrefix = "[B]";
    private const string LargePrefix = "[L]";
    private const string CenterPrefix = "[C]";
    private const string RightPrefix = "[R]";
    private const string LeftRightPrefix = "[LR]";
    private const char LeftRightSep = '|';

    public static ReceiptDocument Parse(string text, string? logoPath = null)
    {
        var lines = text
            .Replace("\r\n", "\n", StringComparison.Ordinal)
            .Replace("\r", "\n", StringComparison.Ordinal)
            .Split('\n')
            .Select(ParseLine)
            .ToList();

        return new ReceiptDocument(lines, logoPath);
    }

    public static ReceiptLine ParseLine(string raw)
    {
        var s = raw;
        var bold = false;
        var large = false;
        var center = false;
        var right = false;
        var isLR = false;
        var leftText = string.Empty;
        var rightText = string.Empty;

        if (s.StartsWith(BoldPrefix, StringComparison.Ordinal))
        {
            bold = true;
            s = s.Substring(BoldPrefix.Length);
        }

        if (s.StartsWith(LargePrefix, StringComparison.Ordinal))
        {
            large = true;
            s = s.Substring(LargePrefix.Length);
        }

        if (s.StartsWith(LeftRightPrefix, StringComparison.Ordinal))
        {
            isLR = true;
            var rest = s.Substring(LeftRightPrefix.Length);
            var parts = rest.Split(LeftRightSep, 2);
            leftText = parts.Length > 0 ? parts[0].Trim() : string.Empty;
            rightText = parts.Length > 1 ? parts[1].Trim() : string.Empty;
            s = string.Empty;
        }
        else if (s.StartsWith(CenterPrefix, StringComparison.Ordinal))
        {
            center = true;
            s = s.Substring(CenterPrefix.Length);
            if (s.StartsWith(LargePrefix, StringComparison.Ordinal))
            {
                large = true;
                s = s.Substring(LargePrefix.Length);
            }
        }
        else if (s.StartsWith(RightPrefix, StringComparison.Ordinal))
        {
            right = true;
            s = s.Substring(RightPrefix.Length);
            if (s.StartsWith(LargePrefix, StringComparison.Ordinal))
            {
                large = true;
                s = s.Substring(LargePrefix.Length);
            }
        }

        return new ReceiptLine(s, bold, large, center, right, isLR, leftText, rightText);
    }
}
